tests/v2/validation/configmaps/generated_fields_test.go

```diff
@@ -0,0 +1,174 @@
+		Labels          map[string]string `yaml:"labels"`
+		Name            string            `yaml:"name"`
+		Namespace       string            `yaml:"namespace"`
+		ResourceVersion string            `yaml:"resourceVersion"`
+		UID             string            `yaml:"uid"`
+	} `yaml:"metadata"`
+}
+
+func (c *ConfigMapTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *ConfigMapTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+	client, err := rancher.NewClient("", c.session)
+	require.NoError(c.T(), err)
+	c.client = client
+	c.steveClient = client.Steve
+	c.httpClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+}
+
+func (c *ConfigMapTestSuite) TestSteveGeneratedFields() {
+
+	configData := map[string]string{
+		"env": "qa",
+	}
+	labels, annotations := map[string]string{}, map[string]string{}
+	configMapName := namegenerator.AppendRandomString("test-configmap")
+	configmap, err := configmaps.CreateConfigMap(c.client, c.client.RancherConfig.ClusterName, configMapName, "auto-generated", "default", configData, labels, annotations)
+	require.NoError(c.T(), err)
+
+	logrus.Infof("created a configmap(%v, configmap).............", configmap.Name)
+
+	c.configMapPayload = &configMapPayload{}
+	copyConfigMapToPayload(configmap, c.configMapPayload)
+
+	// Update the configmap payload with the fields that we want to test.
+	c.configMapPayload.Metadata.Fields = "test-fields"
+	c.configMapPayload.Metadata.Relationships = "test-relationships"
+	c.configMapPayload.Metadata.State = "test-state"
+	c.configMapPayload.Data["foo"] = "bar"
+
+	headers, _, err := postAction(c, configmap.Namespace, configmap.Name)
+	require.NoError(c.T(), err)
+
+	// Check for warnings in the response headers.
+	warnings, ok := headers["Warning"]
+	var failWarnings []string
+	if ok {
+		for _, warning := range warnings {
+			// If the warning message starts with "299", then log it and add it to failWarnings.
+			if strings.HasPrefix(warning, "299") {
+				logrus.Printf("Warning header found: %s", warning)
+				failWarnings = append(failWarnings, warning)
+			}
+		}
+	}
+
+	// If there were any "299" warnings, fail the test.
+	if len(failWarnings) > 0 {
+		require.Fail(c.T(), fmt.Sprintf("Test failed due to warnings: \n%s", strings.Join(failWarnings, "\n")))
+	}
+}
+
+func TestConfigMapTestSuite(t *testing.T) {
+	suite.Run(t, new(ConfigMapTestSuite))
+}
+
+func copyConfigMapToPayload(cm *v1.ConfigMap, payload *configMapPayload) {
+	payload.APIVersion = APIVersion
+	payload.Data = cm.Data
+	payload.Kind = Kind
+	payload.Metadata.Name = cm.ObjectMeta.Name
+	payload.Metadata.Namespace = cm.ObjectMeta.Namespace
+	payload.Metadata.ResourceVersion = cm.ObjectMeta.ResourceVersion
+	payload.Metadata.UID = string(cm.ObjectMeta.UID)
+}
+
+func postAction(c *ConfigMapTestSuite, namespace, name string) (http.Header, *v1.ConfigMap, error) {
```

Comment:
Is this the function we were talking about for checking the response header? Can we move this to the framework to be used as a generic command? Since this is a Steve call we could even make it a part of the client, so we can have some reusability.

Reply:
Yeah this was the part we talked about using `net/http`. Ah okay got it, will work on moving it to the framework so there is reusability. Thanks!

Reply:
Updated to a more reusable method on the client